### ISSUE TITLE
`approx_const` not MSRV aware

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#7623]
Lint name: approx_const

I tried this code:

```rust
#![feature(custom_inner_attributes)]
#![clippy::msrv = "1.30.0"]

fn main() {
    let _log2_10 = 3.321928095f64; 
    let _log10_2 = 0.301029995f64;
}
```

I expected to see this happen:
No warnings. The `LOG2_10`  and `LOG10_2` consts where added in 1.43.

Instead, this happened:
```
error: approximate value of `f64::consts::LOG2_10` found. Consider using it directly
 --> src/main.rs:5:20
  |
5 |     let _log2_10 = 3.321928095f64; 
  |                    ^^^^^^^^^^^^^^
  |
  = note: `#[deny(clippy::approx_constant)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#approx_constant

error: approximate value of `f64::consts::LOG10_2` found. Consider using it directly
 --> src/main.rs:6:20
  |
6 |     let _log10_2 = 0.301029995f64;
  |                    ^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#approx_constant
```
